plugins/volumeFilter_test.go

```diff
@@ -329,3 +328,84 @@ func makeManageSellOffer(price string, amount string) *txnbuild.ManageSellOffer
 		Amount:  amount,
 	}
 }
+
+func TestValidateConfig(t *testing.T) {
+	testCases := []struct {
+		name         string
+		baseCapBase  *float64
+		baseCapQuote *float64
+		marketIDs    []string
+		accountIDs   []string
+		wantErr      error
+	}{
+		{
+			name:         "success - base",
+			baseCapBase:  pointy.Float64(1.0),
+			baseCapQuote: nil,
+			marketIDs:    []string{},
+			accountIDs:   []string{},
+			wantErr:      nil,
+		},
+		{
+			name:         "success - quote",
+			baseCapBase:  nil,
+			baseCapQuote: pointy.Float64(1.0),
+			marketIDs:    []string{},
+			accountIDs:   []string{},
+			wantErr:      nil,
+		},
+		{
+			name:         "failure - 2 nil caps",
+			baseCapBase:  nil,
+			baseCapQuote: nil,
+			marketIDs:    []string{},
+			accountIDs:   []string{},
+			wantErr:      fmt.Errorf("invalid asset caps: only one asset cap can be non-nil, but both are nil"),
+		},
+		{
+			name:         "failure - 2 non-nil caps",
+			baseCapBase:  pointy.Float64(1.0),
+			baseCapQuote: pointy.Float64(1.0),
+			marketIDs:    []string{},
+			accountIDs:   []string{},
+			wantErr:      fmt.Errorf("invalid asset caps: only one asset cap can be non-nil, but both are non-nil"),
+		},
+		{
+			name:         "failure - nil market IDs",
+			baseCapBase:  pointy.Float64(1.0),
+			baseCapQuote: nil,
+			marketIDs:    nil,
+			accountIDs:   []string{},
+			wantErr:      fmt.Errorf("invalid market ids: must be non-nil"),
+		},
+		{
+			name:         "failure - nil account IDs",
+			baseCapBase:  pointy.Float64(1.0),
+			baseCapQuote: nil,
+			marketIDs:    []string{},
+			accountIDs:   nil,
+			wantErr:      fmt.Errorf("invalid optional account ids: must be non-nil"),
+		},
+	}
+
+	for _, mode := range []volumeFilterMode{volumeFilterModeExact, volumeFilterModeIgnore} {
```

Comment:
Those changes are already done. It now loops over both buy and sell actions (which is what these comments request). I'm also looping over the mode, as that's the testing pattern we've used in the tests done in the last few weeks - as I put out this PR 2-3 weeks ago, before we started varying over modes and actions in these config tests. But the mode loop is not essential, and I can hard code the mode in the testing struct if you think it makes sense.

I've also added two invalid test cases - one to test invalid action, one to test invalid mode, since those wouldn't be caught by the above testing. 2 versus 1 failure cases isn't a huge difference, especially when both test otherwise untested failure cases.

That said, this is making me reconsider the quantitative checks of red and green lines outlined in the PR policy we're discussing. The vast majority of the red and green lines in this file are because of a level of indentation changing. This is directly because of the request to test buy and sell actions for most test cases. It's just easier to do that systematically than pick some successful cases and create a couple of test cases. The additional changes due to mode are (+1, -10): one additional line to loop over the mode, and ten deletions of the mode from each test. 

As discussed above, the other changes displayed are the result of separate changes that came up when I ran these tests locally, found a bunch of tests now failing because of validation, and fixed those. I'm not sure why the CI didn't trigger those failures, since validate was being called before. But I don't think not including them in this PR is an option. Agreed the process level fix here is to include a comment about this - which I sent over Slack but somehow didn't get commented to the PR - that's on me, so all good there.

Reply:
Yup, perfect.